app-backend/common/src/main/scala/package.scala

```diff
@@ -56,4 +63,216 @@ package object common extends GeoJsonImplicits {
+  implicit val tiffTagsDecoder: Decoder[TiffTags] = deriveDecoder[TiffTags]
+
+  implicit val storageMethodDecoder: Decoder[StorageMethod] =
+    new Decoder[StorageMethod] {
+      final def apply(c: HCursor): Decoder.Result[StorageMethod] = {
+        for {
+          storageType <- c.downField("storageType").as[String]
+        } yield {
+          storageType match {
+            case "striped" => {
+              val rowsPerStrip =
+                c.downField("rowsPerStrip").as[Option[Int]] match {
+                  case Left(_)  => None
+                  case Right(v) => v
+                }
+              new Striped(rowsPerStrip)
+            }
+            case _ => {
+              val cols = c.downField("cols").as[Int] match {
+                case Left(_)  => 256
+                case Right(v) => v
+              }
+              val rows = c.downField("rows").as[Int] match {
+                case Left(_)  => 256
+                case Right(v) => v
+              }
+              Tiled(cols, rows)
+            }
+          }
+        }
+      }
+    }
+
+  implicit val storageMethodEncoder: Encoder[StorageMethod] =
+    new Encoder[StorageMethod] {
+      final def apply(a: StorageMethod): Json = a match {
+        case _: Striped => Json.obj(("storageType", Json.fromString("striped")))
+        case Tiled(cols, rows) =>
+          Json.obj(("storageType", Json.fromString("tiled")),
+                   ("cols", Json.fromInt(cols)),
+                   ("rows", Json.fromInt(rows)))
+      }
+    }
+
+  implicit val compressionDecoder: Decoder[Compression] =
+    new Decoder[Compression] {
+      final def apply(c: HCursor): Decoder.Result[Compression] = {
+        for {
+          compressionType <- c.downField("compressionType").as[String]
+        } yield {
+          compressionType match {
+            case "NoCompression" => NoCompression
+            case _ => {
+              c.downField("level").as[Int] match {
+                case Left(_)  => DeflateCompression()
+                case Right(i) => DeflateCompression(i)
+              }
+            }
+          }
+        }
+      }
+    }
+
+  implicit val compressionEncoder: Encoder[Compression] =
+    new Encoder[Compression] {
+      final def apply(a: Compression): Json = a match {
+        case NoCompression =>
+          Json.obj(("compressionType", Json.fromString("NoCompression")))
+        case d: DeflateCompression =>
+          Json.obj(("compressionType", Json.fromString("Deflate")),
+                   ("level", Json.fromInt(d.level)))
+      }
+    }
+
+  implicit val indexedColorMapDecoder: Decoder[IndexedColorMap] =
+    Decoder.decodeSeq[Int].emap { s =>
+      Either.right(new IndexedColorMap(s))
+    }
+  implicit val indexedColorMapEncoder: Encoder[IndexedColorMap] =
+    Encoder.encodeSeq[Int].contramapArray(_.colors.toSeq)
+
+  implicit val interLeaveMethodDecoder: Decoder[InterleaveMethod] =
+    deriveDecoder[InterleaveMethod]
```

Comment:
This is potentially not what you wanted, I think it'll work, just not the way you expected:

```scala
@ implicit val en = deriveEncoder[InterleaveMethod]
en: ObjectEncoder[InterleaveMethod] = io.circe.generic.encoding.DerivedObjectEncoder$$anon$1@72d7c7d9

@ import io.circe.syntax._
import io.circe.syntax._

@ PixelInterleave.asInstanceOf[InterleaveMethod].asJson
res12: Json = JObject(
  object[PixelInterleave -> {

}]
)

@ BandInter.asInstanceOf[InterleaveMethod].asJson
BandInterleave
@ BandInterleave.asInstanceOf[InterleaveMethod].asJson
res14: Json = JObject(
  object[BandInterleave -> {

}]
)

``

Reply:
Ah, good catch. I switched this up.